utils/nctl/sh/scenarios/upgrade_after_emergency_upgrade_test.sh

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env bash
+
+source "$NCTL"/sh/utils/main.sh
+source "$NCTL"/sh/views/utils.sh
+source "$NCTL"/sh/assets/upgrade.sh
+source "$NCTL"/sh/scenarios/common/itst.sh
+source "$NCTL"/sh/node/svc_"$NCTL_DAEMON_TYPE".sh
+
+# Exit if any of the commands fail.
+set -e
+
+#######################################
+# Runs an integration tests that performs an emergency restart on the network.
+# It also simulates social consensus on replacing the original validators (nodes 1-5)
+# with a completely new set (nodes 6-10).
+#
+# Arguments:
+#   `timeout=XXX` timeout (in seconds) when syncing. Default=300 seconds.
+#   `version=X_Y_Z` new protocol version to upgrade to. Default=2_0_0.
+#######################################
+function main() {
+    log "------------------------------------------------------------"
+    log "Emergency upgrade test begins"
+    log "------------------------------------------------------------"
+
+    do_await_genesis_era_to_complete
+
+    # 1. Send batch of Wasm deploys
+    do_send_wasm_deploys
+    # 2. Send batch of native transfers
+    do_send_transfers
+    # 3. Wait until they're all included in the chain.
+    do_await_deploy_inclusion
+    # 4. Stop the network for the emergency upgrade.
+    do_stop_network
+    # 5. Prepare the nodes for the emergency upgrade.
+    do_prepare_upgrade
+    # 6. Restart the network (start both the old and the new validators).
+    do_restart_network
+    # 7. Wait for the network to upgrade.
+    do_await_network_upgrade
+    # 8. Send batch of Wasm deploys
+    do_send_wasm_deploys
+    # 9. Send batch of native transfers
+    do_send_transfers
+    # 10. Wait until they're all included in the chain.
+    do_await_deploy_inclusion
+    # 11. Prepare the second upgrade.
+    do_upgrade_second_time
+    # 12. Await the second upgrade.
+    do_await_second_upgrade
+    # 13. Reset node 1 to sync from scratch.
+    do_reset_node_1
+    # 14. Wait until node 1 syncs back to genesis.
+    do_await_node_1_historical_sync
+    # 15. Run Health Checks
+    # ... restarts=15: due to node being stopped and started
+    # ... crashes=5: expected in an emergency restart scenario?
```

Comment:
Good catch, although it's not the comment that needs to be fixed :sweat_smile: But rather the checks below.
There will be multiple restarts detected, probably 16, actually: nodes 2-5 will have restarted twice, and nodes 6-10 once. Node 1 might count for 3 restarts because of the last one with clearing its state.
I'll have to double check what counts as a restart before changing this part.

Reply:
Ah, right. I forgot that the test actually doesn't pass yet, so the counters might change :)